pkg/render/intrusion_detection_test.go

```diff
@@ -361,31 +396,35 @@ var _ = Describe("Intrusion Detection rendering tests", func() {
 		dp := rtest.GetResource(resources, "intrusion-detection-controller", "tigera-intrusion-detection", "apps", "v1", "Deployment").(*appsv1.Deployment)
 		envs := dp.Spec.Template.Spec.Containers[0].Env
 
-		expectedEnvs := []struct {
-			name       string
-			val        string
-			secretName string
-			secretKey  string
-		}{
+		expectedEnvs := []expectedEnvVar{
 			{"CLUSTER_NAME", cfg.ESClusterConfig.ClusterName(), "", ""},
 			{"MULTI_CLUSTER_FORWARDING_CA", cfg.TrustedCertBundle.MountPath(), "", ""},
 			{"IDS_ENABLE_EVENT_FORWARDING", "true", "", ""},
 		}
-		for _, expected := range expectedEnvs {
-			if expected.val != "" {
-				Expect(envs).To(ContainElement(corev1.EnvVar{Name: expected.name, Value: expected.val}))
-			} else {
-				Expect(envs).To(ContainElement(corev1.EnvVar{
-					Name: expected.name,
-					ValueFrom: &corev1.EnvVarSource{
-						SecretKeyRef: &corev1.SecretKeySelector{
-							LocalObjectReference: corev1.LocalObjectReference{Name: expected.secretName},
-							Key:                  expected.secretKey,
-						},
-					},
-				}))
-			}
+
+		assertEnvVarlistMatch(envs, expectedEnvs)
+
+		// expect AD PodTemplate EnvVars
+		expectedADEnvs := []expectedEnvVar{
+			{"ELASTIC_HOST", dns.GetServiceDNSNames(render.ESGatewayServiceName, render.ElasticsearchNamespace, cfg.ClusterDomain)[2], "", ""},
+			{"ELASTIC_PORT", strconv.Itoa(render.ElasticsearchDefaultPort), "", ""},
+			{"ELASTIC_CA", certificatemanagement.TrustedCertBundleMountPath, "", ""},
+			{"ELASTIC_USERNAME", "", render.ElasticsearchADJobUserSecret, "username"},
+			{"ELASTIC_PASSWORD", "", render.ElasticsearchADJobUserSecret, "password"},
+			{"MODEL_STORAGE_API_HOST", dns.GetServiceDNSNames(render.ADAPIObjectName, render.IntrusionDetectionNamespace, cfg.ClusterDomain)[2], "", ""},
```

Comment:
Maybe if you fully spell the value here, we never risk changes to   `dns.GetServiceDNSNames()` to break your feature.